stl/inc/algorithm

```diff
@@ -7404,13 +7404,16 @@ template <class _Diff1, class _Diff2>
 _NODISCARD constexpr auto _Idl_dist_add(_Diff1 _Lhs, _Diff2 _Rhs) {
     (void) _Lhs;
     (void) _Rhs;
-    if constexpr (is_same_v<_Diff1, _Distance_unbounded> || is_same_v<_Diff2, _Distance_unbounded>) {
-        return _Distance_unbounded{};
-    } else if constexpr (is_same_v<_Diff1, _Distance_unknown> || is_same_v<_Diff2, _Distance_unknown>) {
-        return _Distance_unknown{};
-    } else {
-        return _Lhs + _Rhs;
-    }
+#if _HAS_CXX20
+    if constexpr (is_same_v<_Diff1, ranges::_Distance_unbounded> || is_same_v<_Diff2, ranges::_Distance_unbounded>) {
+        return ranges::_Distance_unbounded{};
+    } else
+#endif // _HAS_CXX20
+        if constexpr (is_same_v<_Diff1, _Distance_unknown> || is_same_v<_Diff2, _Distance_unknown>) {

```

Comment:
This `else` is weird - we don't have this pattern anywhere else. If a statement were inserted here, it would lead to non-braced control flow problems. I won't demand a testing reset, but I will clean it up later if you don't get to it first.

Reply:
Pushed a cleanup for this into my `misc` branch.